src/util.cpp

```diff
@@ -723,18 +723,18 @@ void RenameThread(const char* name)
 
 void SetupEnvironment()
 {
-#ifndef WIN32
-    try
-    {
-#if BOOST_FILESYSTEM_VERSION == 3
-            boost::filesystem::path::codecvt(); // Raises runtime error if current locale is invalid
```

Comment:
Are you sure this will still work for boost filesystem v3? I'd assume this `#ifdef` construction was added for a reason.